vital/algo/resection_camera.h

```diff
@@ -0,0 +1,97 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief resection_camera algorithm definition
+ */
+
+#ifndef VITAL_ALGO_RESECTION_CAMERA_H_
+#define VITAL_ALGO_RESECTION_CAMERA_H_
+
+#include <vital/algo/algorithm.h>
+#include <vital/types/camera_perspective.h>
+#include <vital/types/feature_track_set.h>
+#include <vital/types/landmark_map.h>
+
+#include <vector>
+
+namespace kwiver {
+
+namespace vital {
+
+namespace algo {
+
+/// An abstract base class to resection a camera using 3D feature and point
+/// projection pairs.
+class VITAL_ALGO_EXPORT resection_camera
+  : public kwiver::vital::algorithm_def< resection_camera >
+{
+public:
+  /// \return name of this algorithm
+  static std::string
+  static_type_name() { return "resection_camera"; }
+
+  /// Estimate camera parameters from 3D points and their corresponding
+  /// projections.
+  /// \param [in] pts2d 2d projections of pts3d in the same order as pts3d
+  /// \param [in] pts3d 3d points in a 1-1 correspondence with pts2d
+  /// \param [out] inliers inlier flags for the point pairs
+  /// \param [in] cal initial guess on intrinsic parameters of the camera
+  /// \return estimated camera parameters
+  virtual
+  kwiver::vital::camera_perspective_sptr
+  resection(
+    std::vector< kwiver::vital::vector_2d > const& pts2d,
+    std::vector< kwiver::vital::vector_3d > const& pts3d,
+    std::vector< bool >& inliers,
```

Comment:
You might consider moving this so it is not mixed in with the input parameters.

Reply:
not sure where to move them